R2API/LanguageAPI.cs

```diff
@@ -198,5 +211,221 @@ public static class LanguageAPI {
         internal static Dictionary<string, string> GenericTokens = new Dictionary<string, string>();
 
         internal static Dictionary<string, Dictionary<string, string>> LanguageSpecificTokens = new Dictionary<string, Dictionary<string, string>>();
+        
+        internal static Dictionary<string, string> _originalTokens = new Dictionary<string, string>();
+        public static ReadOnlyDictionary<string, string> OriginalTokens = new ReadOnlyDictionary<string, string>(_originalTokens);
+
+        internal static Dictionary<string, string> GenericOverlays = new Dictionary<string, string>();
+
+        internal static Dictionary<string, Dictionary<string, string>> LanguageSpecificOverlays = new Dictionary<string, Dictionary<string, string>>();
+
+        internal delegate void SetupLanguageOverlays();
+        internal static event SetupLanguageOverlays onSetupLanguageOverlays;
+
+        /// <summary>
+        /// Manages temporary language token changes.
+        /// </summary>
+        public class LanguageOverlay {
+            private readonly OverlayTokenData[] overlays;
+            /// <summary>Contains information about the language token changes this LanguageOverlay makes.</summary>
+            public readonly ReadOnlyCollection<OverlayTokenData> readOnlyOverlays;
+
+            internal LanguageOverlay(OverlayTokenData[] _overlays) {
+                overlays = _overlays;
+                readOnlyOverlays = new ReadOnlyCollection<OverlayTokenData>(overlays);
+            }
+
+            internal LanguageOverlay(OverlayTokenData _singleOverlay) {
+                overlays = new OverlayTokenData[]{_singleOverlay};
+                readOnlyOverlays = new ReadOnlyCollection<OverlayTokenData>(overlays);
+            }
+
+            internal void Add() {
+                onSetupLanguageOverlays += LanguageOverlay_onSetupLanguageOverlays;
+            }
+
+            /// <summary>Undoes this LanguageOverlay's language token changes; you may safely dispose it afterwards. Requires a language reload to take effect.</summary>
+            public void Remove() {
+                onSetupLanguageOverlays -= LanguageOverlay_onSetupLanguageOverlays;
+            }
+
+            private void LanguageOverlay_onSetupLanguageOverlays() {
+                foreach(var overlay in overlays) {
+                    Dictionary<string, string> targetDict;
+                    if(overlay.isGeneric) {
+                        targetDict = GenericOverlays;
+                    } else {
+                        if(!LanguageSpecificOverlays.ContainsKey(overlay.lang))
+                            LanguageSpecificOverlays.Add(overlay.lang, new Dictionary<string, string>());
+                        targetDict = LanguageSpecificOverlays[overlay.lang];
+                    }
+                    if(targetDict.ContainsKey(overlay.key))
```

Comment:
`targetDict[overlay.key] = overlay.value;` will work even if the key isn't contained, so it should just be `targetDict[overlay.key] = overlay.value;`